ynr/apps/people/helpers.py

```diff
@@ -77,28 +80,42 @@ def squash_whitespace(s):
 
 
 def clean_mastodon_username(username):
-    # Accounts can be on any domain name and I expect we can get valid
+    # Accounts can be on any domain name and we can expect we can get valid
     # input in at least two formats: @username@domain.com or domain.com/@username.
-    # Both are valid, but to link to the account we're going to need the latter.
-    if re.sub(r"^.*mastodon\.(me.uk|social)/@", "", username):
-        return username
-    if username.count("@") > 1:
-        name = username.split("@")[2]
-        domain = username.split("@")[1]
-        name = name.strip()
-        domain = domain.strip("/")
-        username = "@{domain}/@{name}".format(domain=domain, name=name)
-        return username
-    elif username.count("@") == 1 and username.count("/") == 1:
-        name = username.split("@")[1]
-        domain = username.split("/")[0]
-        domain = domain.strip()
-        name = name.strip()
-        username = "{domain}/@{name}".format(domain=domain, name=name)
-        return username
-    else:
-        message = "The Mastodon username does not appear to be valid. It should be in the format @username@domain.com or domain.com/@username."
+    # Both are valid formats, and to verify the acount is valid, we will need
+    # to extract the domain.
+    parsed_username = urlparse(username)
+    if (
+        not parsed_username.scheme
+        or not parsed_username.netloc
+        or not parsed_username.path
+    ):
+        message = "The Mastodon account must follow the format https://domain/@username. The domain can be any valid Mastodon domain name."
         raise ValueError(message)
+    name = parsed_username.path.split("@")[1]
+    domain = parsed_username.netloc
+    name = name.strip()
+    domain = domain.strip()
+    if not verify_mastodon_server(domain):
+        raise ValueError(
+            "The Mastodon domain {domain} is not a valid Mastodon domain. Please check the domain name and try again.".format(
+                domain=domain
```

Comment:
@Bekabyx @symroe I'm struggling to figure out how to handle this case. I want to verify the domain/server before I try to verify the username exists and return an error to the user in the form. This isn't it.  Advice?